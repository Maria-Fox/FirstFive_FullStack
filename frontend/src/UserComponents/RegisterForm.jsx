import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import AlertNotification from "../Common/AlertNotifications";
import {
  Form,
  FormGroup,
  Label,
  Input,
  Card,
  CardBody,
  Button
} from "reactstrap";

const RegisterForm = ({ registerUser }) => {


  // ***************************************************************

  let initial_state = {
    username: null,
    password: null,
    email: null,
    bio: null
  };

  let navigate = useNavigate();

  let [formData, setFormData] = useState(initial_state);
  let [errors, setErrors] = useState();

  // ***************************************************************

  let handleChange = (e) => {
    let { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  };

  // ***************************************************************

  async function handleSubmit(e) {
    e.preventDefault();
    let response = await registerUser(formData);

    if (response.success) {
      navigate("/home");
    } else {
      setErrors([response.errors]);
    };
  };

  // ***************************************************************


  return (
    <div className="container">
      <h2 className="mb-3 text-white mt-5">Register</h2>

      <Card>

        {errors ? <AlertNotification messages={errors} /> : null}

        <CardBody>
          <Form className="container">
            <FormGroup  >
              <Label for="username" className="mt-2">Username
              </Label>
                <Input
                  id="username"
                  name="username"
                  type="text"
                  placeholder="Username"
                  onChange={handleChange} />
            </FormGroup>

            <FormGroup >
              <Label for="password" className="mt-2">Password
              </Label>
                <Input
                  id="password"
                  name="password"
                  type="password"
                  autoComplete="on"
                  placeholder="@ least 6 characters"
                  onChange={handleChange}
                  minLength = "6"
                  className="RegisterInput" />
              
            </FormGroup>

            <FormGroup >
              <Label for="email" className="mt-2">Email
              </Label>
                <Input
                  id="email"
                  name="email"
                  type="text"
                  placeholder="email@aol.com"
                  required
                  onChange={handleChange} />
            </FormGroup>

            <FormGroup >
              <Label for="bio" className="mt-2">Bio
              </Label>
                <Input
                  id="bio"
                  name="bio"
                  type="text"
                  placeholder="CS student"
                  onChange={handleChange} />
            </FormGroup>


            <Button onClick={handleSubmit} >Submit</Button>
          </Form>
        </CardBody>
      </Card>
    </div >
  );
};

export default RegisterForm;