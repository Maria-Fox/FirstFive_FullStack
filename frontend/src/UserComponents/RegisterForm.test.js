import React from "react";
import { render } from "@testing-library/react";
import RegisterForm from "./RegisterForm";
import { MemoryRouter } from "react-router-dom";

test("Renders RegistrationForm componenet", function () {
  render(
    <MemoryRouter>
      <RegisterForm />
    </MemoryRouter>);
});

test("Rendered RegisterForm correctly", function () {

  // as fragment is a method destructured from the return obj. Puts return item in a div and we can test what's in the fragment.
  let { asFragment } = render(
    <MemoryRouter>
      <RegisterForm />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});