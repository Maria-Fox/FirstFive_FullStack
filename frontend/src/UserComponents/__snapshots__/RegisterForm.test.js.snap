// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered RegisterForm correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2
      class="mb-3 text-white mt-5"
    >
      Register
    </h2>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form
          class="container"
        >
          <div
            class="mb-3"
          >
            <label
              class="mt-2 form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="Username"
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="mt-2 form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="on"
              class="RegisterInput form-control"
              id="password"
              minlength="6"
              name="password"
              placeholder="@ least 6 characters"
              type="password"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="mt-2 form-label"
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="email@aol.com"
              required=""
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="mt-2 form-label"
              for="bio"
            >
              Bio
            </label>
            <input
              class="form-control"
              id="bio"
              name="bio"
              placeholder="CS student"
              type="text"
            />
          </div>
          <button
            class="btn btn-secondary"
            type="button"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
